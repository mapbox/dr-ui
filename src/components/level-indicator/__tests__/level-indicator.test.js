import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './level-indicator-test-cases.js';

describe('level-indicator', () => {
  describe(testCases.levelOne.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.levelOne;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe(testCases.levelTwo.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.levelTwo;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
  describe(testCases.levelThree.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.levelThree;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
