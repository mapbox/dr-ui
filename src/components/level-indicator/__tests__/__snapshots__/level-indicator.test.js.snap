// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`level-indicator Basic renders as expected 1`] = `
<div>
  <div
    className="flex-parent flex-parent--center-cross"
  >
    <div
      className="inline-block w6 bg-green align-middle mr3"
      style={
        Object {
          "borderRadius": "1px",
          "height": "8px",
        }
      }
    />
    <div
      className="inline-block w6 bg-gray-light align-middle mr3"
      style={
        Object {
          "borderRadius": "1px",
          "height": "8px",
        }
      }
    />
    <div
      className="inline-block w6 bg-gray-light align-middle mr3"
      style={
        Object {
          "borderRadius": "1px",
          "height": "8px",
        }
      }
    />
    <div
      className="inline-block color-green ml3"
    >
      Beginner
    </div>
  </div>
  <div
    className="flex-parent flex-parent--center-cross"
  >
    <div
      className="inline-block w6 bg-orange align-middle mr3"
      style={
        Object {
          "borderRadius": "1px",
          "height": "8px",
        }
      }
    />
    <div
      className="inline-block w6 bg-orange align-middle mr3"
      style={
        Object {
          "borderRadius": "1px",
          "height": "8px",
        }
      }
    />
    <div
      className="inline-block w6 bg-gray-light align-middle mr3"
      style={
        Object {
          "borderRadius": "1px",
          "height": "8px",
        }
      }
    />
    <div
      className="inline-block color-orange ml3"
    >
      Intermediate
    </div>
  </div>
  <div
    className="flex-parent flex-parent--center-cross"
  >
    <div
      className="inline-block w6 bg-red align-middle mr3"
      style={
        Object {
          "borderRadius": "1px",
          "height": "8px",
        }
      }
    />
    <div
      className="inline-block w6 bg-red align-middle mr3"
      style={
        Object {
          "borderRadius": "1px",
          "height": "8px",
        }
      }
    />
    <div
      className="inline-block w6 bg-red align-middle mr3"
      style={
        Object {
          "borderRadius": "1px",
          "height": "8px",
        }
      }
    />
    <div
      className="inline-block color-red ml3"
    >
      Advanced
    </div>
  </div>
</div>
`;
